From a1b3c807ce33a377403decead610b6674d44aceb Mon Sep 17 00:00:00 2001
From: Anaconda Recipes <build@anaconda.com>
Date: Fri, 11 Sep 2026 11:15:00 -0600
Subject: [PATCH 0047] Fix protobuf7 SerializeToString nodiscard on linux

Protobuf 7 marks MessageLite::SerializeToString as [[nodiscard]]. XLA's
trace_container ignores the return value; GCC with -Werror=unused-result
fails the linux build.
---
 third_party/xla/xla/tsl/profiler/convert/trace_container.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/xla/xla/tsl/profiler/convert/trace_container.cc b/third_party/xla/xla/tsl/profiler/convert/trace_container.cc
--- a/third_party/xla/xla/tsl/profiler/convert/trace_container.cc
+++ b/third_party/xla/xla/tsl/profiler/convert/trace_container.cc
@@ -58,7 +58,7 @@ void TraceContainer::FlushAndSerializeEvents(std::string* output) {
   // buffer, we must clear the buffer, to prevent overreleasing.
   events_.clear();
-  trace.SerializeToString(output);
+  (void)trace.SerializeToString(output);
 }
 
 }  // namespace profiler
